samples/java-eventsourced-customer-registry/src/main/java/customer/view/CustomerByNameView.java

```diff
@@ -8,48 +8,69 @@
 import com.akkaserverless.javasdk.view.ViewContext;
 import com.google.protobuf.Any;
 import customer.domain.CustomerDomain;
+import customer.api.CustomerApi;
 
 public class CustomerByNameView extends AbstractCustomerByNameView { // <1>
 
   public CustomerByNameView(ViewContext context) {}
 
   @Override
-  public CustomerDomain.CustomerState emptyState() { // <2>
+  public CustomerApi.Customer emptyState() { // <2>
     return null;
   }
 
   @Override // <3>
-  public UpdateEffect<CustomerDomain.CustomerState> processCustomerCreated(
-    CustomerDomain.CustomerState state,
+  public UpdateEffect<CustomerApi.Customer> processCustomerCreated(
+    CustomerApi.Customer state,
```

Comment:
One more thing that we could do to avoid confusion is to call this argument `viewState` to emphasize that this is the persisted view representation.

If that was the view from a value entity and combined with `transform_updates=true`, the method signature would be

```java
processCustomerCreated(CustomerApi.Customer state, CustomerApi.Customer customer)
```

And that can lead to confusion which one is the incoming one and each one is the view state.